### PR TITLE
Enhancement: include owner username in post-consumption variables

### PR DESCRIPTION
**Pull request by Freddy-0**
## Proposed change

Add the ability to access the document owner name in the post-consume script.
Useful for sending messages to the user (e.g., ntfy in my case).
Currently, accessing the owner name requires using the API. (IMO overkill for simply getting the user name)
Exposes the owner name as a new variable in the post-consume script.
The owner name is already used in the consumer.py script. Just exposing it via env variable to the post consume script now.

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by Freddy-0** [comment]
Example for sending a message pre change with the api:

```python
#!/usr/bin/env python3

import os, requests, json

DOCUMENT_ID = os.getenv('DOCUMENT_ID')
BASE_URL = "https://documents.higgs-boson.eu"
API_HEADERS = {'Authorization': 'Token REDACTED', 'Cookie': 'csrftoken=REDACTED'}

doc_resp = requests.get(f"{BASE_URL}/api/documents/{DOCUMENT_ID}/", headers=API_HEADERS)
doc_data = doc_resp.json()

user_resp = requests.get(f"{BASE_URL}/api/users/", headers=API_HEADERS)
user_data = user_resp.json()

Benutzer = next((user["username"] for user in user_data["results"] if user["id"] == doc_data['owner']), None)

requests.post("NTFY_URL_REDACTED", data=json.dumps({
    "topic": Benutzer,
    "message": f"Hi, {Benutzer}, ein Dokument wurde hinzugefügt",
    "actions": [{"action": "view", "label": "Öffnen", "url": f"paperparrot://documents/{DOCUMENT_ID}"}]
}), headers={"Click": f"paperparrot://documents/{DOCUMENT_ID}"})
```

Example for sending a message post change without the api:
```Python
#!/usr/bin/env python3

import os, requests, json

DOCUMENT_ID = os.getenv('DOCUMENT_ID')
Benutzer = os.getenv('DOCUMENT_OWNER')
PAPERPARROT_URL = f"paperparrot://documents/{DOCUMENT_ID}"

requests.post("NTFY_URL_REDACTED", data=json.dumps({
    "topic": Benutzer,
    "message": f"Hi, {Benutzer}, ein Dokument wurde hinzugefügt",
    "actions": [{"action": "view", "label": "Öffnen", "url": PAPERPARROT_URL}]
}), headers={"Click": PAPERPARROT_URL})
```

**Comment by stumpylog** [comment]
The owner may not be set (ie null).  Should the value even be provided in that case?

**Comment by Freddy-0** [comment]
I would assume so, because this information is still valid and the user can work with that in the post consumption script.

Even if there is no owner, this information (ie null) might be helpful for the script / logic

**Comment by stumpylog** [comment]
In that case, the documentation should be clear about it.  The value would be the string "None" if no owner.

**Comment by Freddy-0** [comment]
Added additional info in form of a footnote. Would this be enough?

File: docs/advanced_usage.md [hunk 1]
```diff
@@ -187,12 +187,14 @@ variables:
 | `DOCUMENT_THUMBNAIL_PATH`    | Path to the generated thumbnail                |
 | `DOCUMENT_DOWNLOAD_URL`      | URL for document download                      |
 | `DOCUMENT_THUMBNAIL_URL`     | URL for the document thumbnail                 |
-| `DOCUMENT_OWNER`             | Name of the owner of the document              |
+| `DOCUMENT_OWNER`             | Name of the owner of the document[^1]          |
```
Comment (shamoon) [inline docs/advanced_usage.md:190]:
```suggestion
| `DOCUMENT_OWNER`             | Name of the owner of the document (or "None")  |
```

File: docs/advanced_usage.md [hunk 2]
```diff
@@ -187,12 +187,14 @@ variables:
...
 | `DOCUMENT_CORRESPONDENT`     | Assigned correspondent (if any)                |
 | `DOCUMENT_TAGS`              | Comma separated list of tags applied (if any)  |
 | `DOCUMENT_ORIGINAL_FILENAME` | Filename of original document                  |
 | `TASK_ID`                    | Task UUID used to import the document (if any) |
 
+[^1] If there is no owner the variable returns "None" which is likely if there is no other script / flow in place that updates the owner beforehand.
+
```
Comment (shamoon) [inline docs/advanced_usage.md:197]:
```suggestion
```

**Review by shamoon — DISMISSED** [review]
Keep it short and sweet IMHO.

Also, isnt it the _username_? https://github.com/django/django/blob/082fe2b5a83571dec4aa97580af0fe8cf2a5214e/django/contrib/auth/base_user.py#L56

Would call it such in the doc

**Comment by shamoon** [comment]
Also, isnt it the _username_? https://github.com/django/django/blob/082fe2b5a83571dec4aa97580af0fe8cf2a5214e/django/contrib/auth/base_user.py#L56

Would call it such in the doc

**Comment by shamoon** [comment]
Sorry, actually one more note is that this is a little inconsistent with e.g. the TASK_ID which outputs `""` if it's empty. I defer to stumpylog here, and not that I feel very strongly but it might make more sense since as-is the string "None" would have to be evaluated as such

E.g. https://github.com/paperless-ngx/paperless-ngx/blob/1d325c233fe2e2ca96d69188ec445b65e8bf97b2/src/documents/consumer.py#L459

**Comment by Freddy-0** [comment]
Fixed the documentation according to your suggestions

**Comment by Freddy-0** [comment]
> Sorry, actually one more note is that this is a little inconsistent with e.g. the TASK_ID which outputs `""` if it's empty. I defer to stumpylog here, and not that I feel very strongly but it might make more sense since as-is the string "None" would have to be evaluated as such
> 
> E.g.
> 
> https://github.com/paperless-ngx/paperless-ngx/blob/1d325c233fe2e2ca96d69188ec445b65e8bf97b2/src/documents/consumer.py#L459

Not sure about what way to make this consistent.   :shrug:

File: src/documents/consumer.py [hunk 3]
```diff
@@ -450,6 +450,7 @@ def run_post_consume_script(self, document: Document):
             "document-thumb",
             kwargs={"pk": document.pk},
         )
+        script_env["DOCUMENT_OWNER"] = str(document.owner)
```
Comment (stumpylog) [inline src/documents/consumer.py:453]: ` =  document.owner.get_username() if document.owner else ""` would do it, then indicate the ("if any") similar to the task id and tags

Comment (Freddy-0) [reply src/documents/consumer.py:453]: :thumbsup: done

File: docs/advanced_usage.md [hunk 4]
```diff
@@ -187,6 +187,7 @@ variables:
 | `DOCUMENT_THUMBNAIL_PATH`    | Path to the generated thumbnail                |
 | `DOCUMENT_DOWNLOAD_URL`      | URL for document download                      |
 | `DOCUMENT_THUMBNAIL_URL`     | URL for the document thumbnail                 |
+| `DOCUMENT_OWNER`             | Username of the document owner (or "None")     |
```
Comment (shamoon) [inline docs/advanced_usage.md:190]:
```suggestion
| `DOCUMENT_OWNER`             | Username of the document owner (if any)        |
```

**Review by stumpylog — APPROVED** [review]

**Pull request merged**